src/plugins/intel_cpu/src/emitters/snippets/x64/jit_loop_base_emitters.hpp

```diff
@@ -0,0 +1,132 @@
+public:
+    jit_loop_begin_base_emitter(dnnl::impl::cpu::x64::jit_generator_t* h,
+                                dnnl::impl::cpu::x64::cpu_isa_t isa,
+                                const ov::snippets::lowered::ExpressionPtr& expr,
+                                bool is_parallel);
+
+    size_t get_inputs_num() const override {
+        return 0;
+    }
+
+    void set_loop_end_label(const std::shared_ptr<const Xbyak::Label>& label) {
+        m_loop_end_label = label;
+    }
+
+    std::shared_ptr<const Xbyak::Label> get_begin_label() const {
+        return m_loop_begin_label;
+    }
+
+protected:
+    std::shared_ptr<Xbyak::Label> m_loop_begin_label = nullptr;
+    std::shared_ptr<const Xbyak::Label> m_loop_end_label = nullptr;
+    size_t m_wa_increment = 0;
+    size_t m_loop_id_offset = 0;
+    bool m_evaluate_once = false;
+
+    static ov::snippets::lowered::ExpressionPtr get_loop_end_expr(const ov::snippets::lowered::ExpressionPtr& expr);
+
+    void validate_arguments(const std::vector<size_t>& in, const std::vector<size_t>& out) const override;
+
+    /**
+     * @brief Loads work amount (either from runtime arguments for dynamic loops
+     * or uses a compile-time constant for static loops), and jumps to
+     * the loop end label if the work amount is less than the increment, effectively skipping the loop body.
+     *
+     * @param out Vector of output register indices, contining the work amount register
+     * @param is_work_amount_dynamic True if work amount is determined at runtime
+     * (and its value should be read from loop_args), false if known at compile time
+     * @param work_amount_static Compile-time work amount value (used only when is_work_amount_dynamic is false)
+     */
+    void emit_loop_begin_work_amount_check(const std::vector<size_t>& out,
+                                           bool is_work_amount_dynamic,
+                                           int64_t work_amount_static) const;
+
+    void emit_code_impl(const std::vector<size_t>& in_idxs,
+                        const std::vector<size_t>& out_idxs,
+                        const std::vector<size_t>& pool_vec_idxs,
+                        const std::vector<size_t>& pool_gpr_idxs) const override;
+};
+
+class jit_loop_end_base_emitter : public jit_emitter {
+public:
+    jit_loop_end_base_emitter(dnnl::impl::cpu::x64::jit_generator_t* h,
+                              dnnl::impl::cpu::x64::cpu_isa_t isa,
+                              const ov::snippets::lowered::ExpressionPtr& expr,
+                              bool is_parallel);
+
+    size_t get_inputs_num() const override {
+        return 0;
+    }
+
+    static jit_snippets_call_args::loop_args_t compose_loop_args(
+        const std::shared_ptr<ov::snippets::op::LoopEnd>& loop_end);
+
+    // `jit_loop_end_base_emitter` handles manually aux_gpr allocation using `jit_aux_gpr_holder`
+    size_t aux_gprs_count() const override {
+        return 0;
+    }
+
```

Comment:
Same as for `jit_loop_begin_base_emitter`, we can remove it here

Reply:
For loop_end emitters, it makes sense to leave `aux_gprs_count` overridden in the base class since both `jit_parallel_loop_end_emitter` and `jit_loop_end_emitter` reuse it

Reply:
Ok, let's leave as is